docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,105 @@
+[Kubernetes Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) (NP).
+NP allow to control traffic flow at the IP address or port level (OSI layer 3 or 4). If a user workload or a Fybrik 
+previous chained module collocated with the next module in the same Kubernetes cluster, NPs can help to restrict the Fybrik 
+module ingress access. However, when a workload or a previous chained module runs out of the cluster, the NP based 
+isolation is going to be more complicated. The collocated deployment can be based on Kubernetes labels or namespaces, but 
+cross-clusters connections can be based on requester IP addresses, which is not always easy to discover. 
+Therefore, later we will provide other isolation methods as an extension to NP.
+
+### Advantages:
+- does not require any changes in modules implementations, can be implemented in the Fybrik control plane only.
+- can be combined with other isolation methods
+
+### Disadvantages:
+- complicated in cross clusters deployments.
+- NP are implemented by the network plugin. Creating a NetworkPolicy resource without a controller - a Container Network 
+Interface (CNI) that implements it will have no effect. Default Kind deployment which by default uses "kindnetd” doesn’t 
+support NP. It requires installation another CNI, e.g. [Calico](https://github.com/projectcalico/calico). Here is the 
+[instructions](https://alexbrand.dev/post/creating-a-kind-cluster-with-calico-networking/). All K8s clusters in 
+production deployments use CNIs which support NP.
+- requires information about workload deployment, and/or setting labels to the workload
+
+## Implementation
+
+The “from” element of ingress 
+[NetworkPolicySpec](https://kubernetes.io/docs/reference/kubernetes-api/policy-resources/network-policy-v1/#NetworkPolicySpec) 
+allows to restrict incoming network traffic based on “podSelectors” (labels based), and/or 
+“namespaceSelectors” (labels based) and/or “ipBlocks”. The last one permits incoming network traffic from predefined IP 
+address blocks.
+
+Current “FybrikApplicationSpec” has the “[selector](https://fybrik.io/v1.2/reference/crds/#fybrikapplicationspecselector)” 
+element, which is a combination of “clusterName” and “workloadSelector”. Unfortunately, based on this information we can 
+implement only one label based “podSelector”. Furthermore, for the backward compatibility of the current implementation 
+assumes that a Read scenario should have at least one label, which is not always true. 
+
+### Suggetions:
+
+- extend the “FybrikApplicationSpec.selector” element with an array of 
+[NetworkPolicyPeers](https://github.com/kubernetes/api/blob/59fcd23597fd090dba6b7e903eb0a8c9e8efb0a6/networking/v1/types.go#L183)
+this “NetworkPoliciesPeers” array will be an input element for the NP `from`element. This will restrict possible incoming
+connections to the relevant Fybrik module.
```

Comment:
Isn't this in essence putting the responsibility of the isolation implementation on the data user / FybrikApplication creator?  Why should they have to know anything about network policies?

Reply:
They don't have to know about network policies, but they do know where and how they will run workloads. It is similar to the label setting requirement.  All entries are optional, so users do not have to set all options. 
In the current implementation, the `workloadSelector` is much more complicated than just a list of labels.

Reply:
This section is unclear to me.  Can you give some examples?  It's really important that we provide something as simple as possible for the user who is creating the FybrikApplication to indicate the workload.  Let Fybrik discover as much as possible on its own.

Reply:
For example, a user might run a Jupiter notebook on his laptop, outside the Fybrik cluster. Therefore, in order to protect the data plane and allow connectivity only from his laptop, the user can provide the IP address as the input information for the NP creation. 

```yaml
apiVersion: app.fybrik.io/v1beta1
kind: FybrikApplication
metadata:
  name: my-notebook
  labels:
    app: my-notebook
spec:
  selector:
    networkPeers:
      - ipBlock:
            cidr: 172.17.23.5/32
  appInfo:
    intent: Fraud Detection
  data:
    - dataSetID: ${CATALOGED_ASSET}
      requirements:
        interface: 
          protocol: fybrik-arrow-flight
```